bin/client/src/cached_eigenda_provider.rs

```diff
@@ -0,0 +1,83 @@
+use alloy_primitives::Bytes;
+use alloy_rlp::Decodable;
+use async_trait::async_trait;
+use kona_preimage::errors::PreimageOracleError;
+use kona_preimage::CommsClient;
+
+use hokulea_cryptography::witness::EigenDABlobWitness;
+use hokulea_eigenda::BlobInfo;
+use hokulea_eigenda::EigenDABlobProvider;
+use hokulea_proof::eigenda_provider::OracleEigenDAProvider;
+
+use kona_proof::errors::OracleProviderError;
+
+use alloc::sync::Arc;
+use rust_kzg_bn254::srs::SRS;
+use std::sync::Mutex;
+
+/// CachedOracleEigenDAProvider is a wrapper outside OracleEigenDAProvider. Its intended use
+/// case is to fetch all eigenda blobs received during the derivation pipeline. So that it
+/// is able to compute and cache the kzg witnesses, which can be verified inside ZKVM by checking
```

Comment:
Could we create a zkvm crate or somethign of that effect? Would be good to separate the structs and logic that is only used for zkvm vs those that are used for fraud proofs.